lib/class-wp-theme-json-gutenberg.php

```diff
@@ -247,16 +247,7 @@ class WP_Theme_JSON_Gutenberg {
 		'--wp--style--block-gap'     => array( 'spacing', 'blockGap' ),
 		'text-decoration'            => array( 'typography', 'textDecoration' ),
 		'text-transform'             => array( 'typography', 'textTransform' ),
-	);
-
-	/**
-	 * Metadata for style properties that need to use the duotone selector.
-	 *
-	 * Each element is a direct mapping from the CSS property name to the
-	 * path to the value in theme.json & block attributes.
-	 */
-	const DUOTONE_PROPERTIES_METADATA = array(
-		'filter' => array( 'filter', 'duotone' ),
+		'filter'                     => array( 'filter', 'duotone' ),
```

Comment:
I had this separate before to make way for adding a generic filter that can apply to entire blocks or elements.

```json
{
	"styles": {
		"elements": {
			"h1": {
				"filter": {
					"custom": "drop-shadow( 3px )"
				}
			}
		}
	}
}
```

Maybe we could do something like `--wp--style--duotone` for the duotone one that it doesn't overlap with the generic `filter` property?

Reply:
I think the property metadata could absorb the selector bound to each property, which is a piece of configuration that lives in the code. I haven't done this yet to take small steps we can land quickly and because this works for the use cases we have so far. When we do that, a CSS property name can match multiple theme.json properties.